### PR TITLE
fix: Fix NPE

### LINKED ISSUE #1382: NPE Parsing JUnit XML
**Describe the bug**
Error is thrown before matrix or storage bucket URLs are printed
Stack trace:
```RunTests
java.lang.NullPointerException
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromString(StdDeserializer.java:267)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromString(BeanDeserializerBase.java:1480)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:207)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:197)
        at com.fasterxml.jackson.dataformat.xml.deser.WrapperHandlingDeserializer.deserialize(WrapperHandlingDeserializer.java:118)
        at com.fasterxml.jackson.dataformat.xml.deser.XmlDeserializationContext.readRootValue(XmlDeserializationContext.java:91)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4591)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3546)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3514)
        at ftl.reports.xml.JUnitXmlKt.parseAllSuitesXml(JUnitXml.kt:59)
        at ftl.reports.xml.JUnitXmlKt.parseAllSuitesXml(JUnitXml.kt:51)
        at ftl.gc.GcStorage.downloadJunitXml(GcStorage.kt:120)
        at ftl.args.ArgsHelper.calculateShards(ArgsHelper.kt:250)
        at ftl.run.platform.android.CreateAndroidTestContextKt.calculateShards(CreateAndroidTestContext.kt:55)
        at ftl.run.platform.android.CreateAndroidTestContextKt.access$calculateShards(CreateAndroidTestContext.kt:1)
        at ftl.run.platform.android.CreateAndroidTestContextKt$setupShards$2$invokeSuspend$$inlined$map$lambda$1.invokeSuspend(CreateAndroidTestContext.kt:41)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:84)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at ftl.cli.firebase.test.android.AndroidRunCommand.run(AndroidRunCommand.kt:48)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1919)
        at picocli.CommandLine.access$1200(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
        at picocli.CommandLine.execute(CommandLine.java:2058)
        at ftl.Main$Companion$main$1.invoke(Main.kt:59)
        at ftl.Main$Companion$main$1.invoke(Main.kt:49)
        at ftl.run.exception.ExceptionHandlerKt.withGlobalExceptionHandling(ExceptionHandler.kt:11)
        at ftl.Main$Companion.main(Main.kt:55)
        at ftl.Main.main(Main.kt)```
```

**To Reproduce**
> Steps to reproduce the behavior:
Only failing for tablet runs:
```
device:
--
  | - model: Nexus9
  | version: 25
  | locale: en
  | orientation: portrait
```

**Expected behavior**
> A clear and concise description of what you expected to happen.
No crash

**Details (please complete the following information):**
> Have you tested on the latest Flank snapshot?
>
> Post the output of ` flank --version`.

I'm using a locally built version at this commit: https://github.com/Flank/flank/commit/674ca739419d5e0382abe47327affb3c6394c344

```
version: local_snapshot
--
  | revision: 8cc0786b203e2b3fa9a998a1c30fef244fc2f3c4

```

**Additional context**
> Add any other context about the problem here.
Started happening sometime between 3:30-4:00 PM Dec 7 2020, America/Los_Angeles

### PR DESCRIPTION
**Pull request by pawelpasterz**
Fixes #1382 

TLDR;
* fix NPE when `<testsuites/>` was deserialized (a workaround was needed though)
* prevent XML results from being uploaded (for smart flank) if contain no results

When an infrastructure error occurs result matrix is empty which leads to empty `JUnitTestResult`being created. Flank should not upload empty results to prevent old data corruption. 

Flank should also handle XML results with a self-closed root element (XML results for smart flank can be also uploaded manually). That could be easily fixed by changing `EMPTY_ELEMENT_AS_NULL` to false but flank uses this configuration quite extensively (ex skipped tag), so it's not an option. A small workaround was implemented: when the XML file is parsed flank checks if the string contains `<testsuites/>`. If does, an empty `JUnitTestResult` object is returned.

Unfortunately, I did not come up with any way of testing it (no idea how to simulate infrastructure error). I covered it in unit test (with mocks)

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

**Review by Sloox — APPROVED** [review]

**Review by adamfilipow92 — APPROVED** [review]
- [X] Tests and checks pass
- [X] Code looks good!
